api/http/handler/webhooks/webhook_execute.go

```diff
@@ -0,0 +1,68 @@
+package webhooks
+
+import (
+	"context"
+	"net/http"
+
+	dockertypes "github.com/docker/docker/api/types"
+	docker "github.com/docker/docker/client"
+	"github.com/portainer/portainer"
+	httperror "github.com/portainer/portainer/http/error"
+	"github.com/portainer/portainer/http/request"
+)
+
+// Acts on a passed in token UUID to restart the docker service
+func (handler *Handler) webhookExecute(w http.ResponseWriter, r *http.Request) *httperror.HandlerError {
+
+	webhookToken, err := request.RetrieveRouteVariableValue(r, "token")
+
+	if err != nil {
+		return &httperror.HandlerError{http.StatusInternalServerError, "Invalid service id parameter", err}
+	}
+
+	webhook, err := handler.WebhookService.WebhookByToken(webhookToken)
+	if err != nil && err != portainer.ErrObjectNotFound {
+		return &httperror.HandlerError{http.StatusInternalServerError, "Unable to retrieve webhook from the database", err}
+	}
+
+	serviceID := webhook.ServiceID
+	endpointID := webhook.EndpointID
+
+	endpoint, err := handler.EndpointService.Endpoint(portainer.EndpointID(endpointID))
+	if err == portainer.ErrObjectNotFound {
+		return &httperror.HandlerError{http.StatusNotFound, "Unable to find an endpoint with the specified identifier inside the database", err}
+	} else if err != nil {
+		return &httperror.HandlerError{http.StatusInternalServerError, "Unable to find an endpoint with the specified identifier inside the database", err}
+	}
+
+	var dockerClient *docker.Client
```

Comment:
The `api/docker` package exposes a `ClientFactory` to create a Docker client from an existing endpoint. You can inject the ClientFactory in this handler and use it instead of re-creating the client manually.